tests/backup/backup_test.go

```diff
@@ -246,71 +249,189 @@ var _ = Describe("{BasicBackupCreation}", func() {
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
 		}
-		dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
-	}
-	log.InfoD("Deleting bucket,backup location and cloud setting")
-	for i, provider := range providers {
-		bucketName := fmt.Sprintf("%s-%s", "bucket", provider)
-		DeleteBucket(provider, bucketName)
-		CredName := fmt.Sprintf("%s-%s", "cred", provider)
-		DeleteCloudCredential(CredName, orgID, CloudCredUID_list[i])
-		// Need to add backup location UID for Delete Backup Location call
-		DeleteBackupLocation(backupLocation, "", orgID)
-	}
-	DeleteCluster(destinationClusterName, OrgID)
-	DeleteCluster(SourceClusterName, OrgID)
-	if flag == false {
-		return false
-	}
-	return true
+	})
+	It("Different Access Same User", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
+		dash.VerifyFatal(err, nil, "Getting context")
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications ")
+			ValidateApplications(contexts)
+		})
+		Step("Create Users", func() {
+			log.InfoD("Creating users testuser")
+			userName = "testuser"
+			firstName := "FirstName"
+			lastName := "LastName"
+			email := "testuser@cnbu.com"
+			err := backup.AddUser(userName, firstName, lastName, email, "Password1")
+			log.FailOnError(err, "Failed to create user - %s", userName)
 
-}
+		})
+		Step("Create Groups", func() {
+			log.InfoD("Creating group testGroup")
+			groupName = fmt.Sprintf("testGroup")
+			err := backup.AddGroup(groupName)
+			log.FailOnError(err, "Failed to create group - %v", groupName)
+
+		})
+		Step("Add users to group", func() {
+			log.InfoD("Adding user to groups")
+			err := backup.AddGroupToUser(userName, groupName)
+			dash.VerifyFatal(err, nil, "Adding user to group")
+			usersOfGroup, err := backup.GetMembersOfGroup(groupName)
+			log.FailOnError(err, "Error fetching members of the group - %v", groupName)
+			log.Infof("Group [%v] contains the following users: \n%v", groupName, usersOfGroup)
+
+		})
+		Step("Creating bucket,backup location and cloud setting", func() {
+			log.InfoD("Creating bucket,backup location and cloud setting")
+			bucketNames := getBucketName()
+			providers := getProviders()
+			for _, provider := range providers {
+				bucketName := fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				cloudCredName := fmt.Sprintf("%s-%s", "cred", provider)
+				bkpLocationName := fmt.Sprintf("%s-%s-bl", provider, bucketNames[0])
+				cloudCredUID := uuid.New()
+				backupLocationUID := uuid.New()
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				time.Sleep(time.Minute * 3)
```

Comment:
seems this wait is being used everytime CreateCloudCredential is called. can you move this into that method itself, so that once we remove this , only one line removal will be enough.

Reply:
done